qa/multi-cluster-search/src/test/java/org/elasticsearch/search/CCSDuelIT.java

```diff
@@ -789,20 +832,144 @@ private static void duelSearch(SearchRequest searchRequest, Consumer<SearchRespo
             }
             SearchResponse minimizeRoundtripsSearchResponse = minimizeRoundtripsResponse.get();
+
             responseChecker.accept(minimizeRoundtripsSearchResponse);
             assertEquals(3, minimizeRoundtripsSearchResponse.getNumReducePhases());
             SearchResponse fanOutSearchResponse = fanOutResponse.get();
             responseChecker.accept(fanOutSearchResponse);
             assertEquals(1, fanOutSearchResponse.getNumReducePhases());
             Map<String, Object> minimizeRoundtripsResponseMap = responseToMap(minimizeRoundtripsSearchResponse);
             Map<String, Object> fanOutResponseMap = responseToMap(fanOutSearchResponse);
-            if (minimizeRoundtripsResponseMap.equals(fanOutResponseMap) == false) {
-                NotEqualMessageBuilder message = new NotEqualMessageBuilder();
-                message.compareMaps(minimizeRoundtripsResponseMap, fanOutResponseMap);
-                throw new AssertionError("Didn't match expected value:\n" + message);
-            }
+            compareResponseMaps(minimizeRoundtripsResponseMap, fanOutResponseMap, "Comparing sync_search minimizeRoundTrip vs. fanOut");
             assertThat(minimizeRoundtripsSearchResponse.getSkippedShards(), lessThanOrEqualTo(fanOutSearchResponse.getSkippedShards()));
+            return minimizeRoundtripsResponseMap;
+        }
+    }
+
+    /**
+     * @return reponseMap from one of the async searches
+     */
+    private static Map<String, Object> duelSearchAsync(SearchRequest searchRequest, Consumer<SearchResponse> responseChecker)
+        throws Exception {
+        searchRequest.setCcsMinimizeRoundtrips(true);
+        AsyncSearchResponse minimizeRoundtripsResponse = submitAsyncSearch(
+            searchRequest,
+            TimeValue.timeValueSeconds(1),
+            restHighLevelClient.getParserConfig()
+        );
+        SearchResponse searchResponse = minimizeRoundtripsResponse.getSearchResponse();
+
+        try {
+            final String responseId = minimizeRoundtripsResponse.getId();
+            assertBusy(() -> {
+                AsyncSearchResponse resp = getAsyncSearch(responseId, restHighLevelClient.getParserConfig());
+                assertThat(resp.isRunning(), equalTo(false));
+            });
+            minimizeRoundtripsResponse = getAsyncSearch(responseId, restHighLevelClient.getParserConfig());
+        } finally {
+            deleteAsyncSearch(minimizeRoundtripsResponse.getId());
+        }
+
+        searchRequest.setCcsMinimizeRoundtrips(false);
+        AsyncSearchResponse fanOutResponse = submitAsyncSearch(
+            searchRequest,
+            TimeValue.timeValueSeconds(1),
+            restHighLevelClient.getParserConfig()
+        );
+        try {
+            final String responseId = fanOutResponse.getId();
+            assertBusy(() -> {
+                AsyncSearchResponse resp = getAsyncSearch(responseId, restHighLevelClient.getParserConfig());
```

Comment:
out of curiosity: you could have cut corners here, and set a high enough timeout for submitAsyncSearch so that you never have to call get and delete. What made you go this route instead?

Reply:
Since users doing CCS will frequently have queries that run longer than 1s, I thought it best to include retrieving the results from the GET endpoint rather than a synchronous return of the POST call.

Ideally I'd like the test to sometimes return synchronously (within the timeout) and sometimes run longer than the timeout, but that would add even more complexity to the test, so I didn't varying the time.